### DISCUSSION TITLE
Golang - Sharing Common Logic / Types

### DISCUSSION THREAD

**Discussion by Incisive** [windmill-labs/windmill#3426] [Questions] [answered]
When importing relative "scripts", I know you can do this with JS languages and seems to function pretty well (except no typehinting which is the only downside) but I didn't see any documentation or examples on doing this with Golang here https://www.windmill.dev/docs/advanced/sharing_common_logic or any issues/discussions on it in this repo.

Is there any support for relative importing of other golang scripts ? I'm trying to come up with a clean way to organize my struct definitions without having to cram it all into a single windmill script, and then duplicate it to the downstream steps.

thanks!

**Comment by rubenfiszel** [comment, marked as answer]
Hi @Incisive ,

Actually with the latest bun improvements, you should see typehinting when using relative imports with `./` or `../`

There are no support for relative importing in golang, probably not because it's impossible but by lack of time to implement it, my lack of expertise in go, and the lack of enough important demand from our enterprise customers to prioritize it. But if we see a surge in demand from our customers, we will prioritize it

**Reply by Incisive** [reply]
Understood, thanks for confirming and also the note about JS typehinting :)!
